contracts/strategies/88mph/base/EightyEightMphStrategy.sol

```diff
@@ -0,0 +1,676 @@
+     *      is not salvagable.
+     */
+    function salvage(
+        address recipient,
+        address token,
+        uint256 amount
+    ) external onlyControllerOrGovernance {
+        // Make sure governance or controller cannot come in and take away the invested tokens
+        require(
+            !unsalvagableTokens(token),
+            "Token is defined as non-salvagable"
+        );
+        IERC20(token).safeTransfer(recipient, amount);
+    }
+
+    function depositArbCheck() public view returns (bool) {
+        return true;
+    }
+
+    function doHardWork() external onlyNotPausedInvesting restricted {
+        // fail as early as possible if deposit must be rolled over
+        uint64 depositMaturity = depositMaturity();
+        if(depositMaturity != 0 && block.timestamp > depositMaturity) {
+            // time to roll over the deposit to a new depositId with extended maturity
+            if(!shouldRolloverDeposit) {
+                // we revert here because we want this process to be kicked off manually through setting the flag
+                // it is imperative that the withdrawal fee is waived for the new depositId - which we can not do with code
+                // If that is not done, this strategy will start losing money for depositors.
+                revert("Deposit must be rolled over. ATTENTION: ENSURE WITHDRAWAL FEE IS WAIVED FOR NEW DEPOSITID");
+            } else {
+                rolloverDeposit(true);
+            }
+        } else {
+            claimRewards();
+            _liquidateReward();
+        }
+        
+        investAllUnderlying();
+    }
+
+    // ---------------- Governance-only methods ----------------
+
+    /**
+     * @dev In case there are issues with the pool or underlying asset,
+     *      governance can still trigger an emergency exit, in order to
+     *      pause investing in the pool.
+     */
+    function emergencyExit() public onlyGovernance {
+        _setPausedInvesting(true);
+        emergencyExitRewardPool();
+    }
+
+    function emergencyExitRewardPool() public onlyGovernance {
+        // don't claim rewards, just withdraw. Use maxInt to withdraw all
+        uint256 maxInt = 2**256 - 1; // see https://forum.openzeppelin.com/t/using-the-maximum-integer-in-solidity/3000
+        bool early = block.timestamp < depositMaturity(); // withdrawing before or after maturation​
+        IDInterest(rewardPool()).withdraw(depositId, maxInt, early);
+    }
+
+    /**
+     * Use to manually rollover a deposit to a new maturity date.
+     * This will create a new deposit using the principal + fixed-rate yield of the old deposit.
+     * doHardWork can be triggered afterwards again if it failed previously because maturity was about to be reached
+     */
+    function rolloverDeposit(bool confirmWithdrawalFeeWillBeWaived) public restricted {
+        // Attention: it is imperative that 88mph waives the early withdrawal fee for the new depositID!
```

Comment:
This method is important, please take note
Please take note about the deal with 88mph that they will waive the early withdrawal fee - see the test for an example

Deposits that have reached maturity must be rolled over. 
But there are a few caveats, such as rolling over does not work if the deposit does not hold any funds etc. 
See the comments in the code for more details.